src/coq/constants/equtils.ml

```diff
@@ -163,3 +163,44 @@ let is_rewrite_r (trm : types) : bool =
  *)
 let is_rewrite (trm : types) : bool =
   is_rewrite_l trm || is_rewrite_r trm
+
+
+(* Information required to perform a rewrite. *)
+type rewrite_args = {
```

Comment:
This one is harder to unify since it's more generic, so I would just include a comment at the interface level explaining it's more generic whereas the other one handles only prop